aiorospy/src/aiorospy/action.py

```diff
@@ -228,19 +230,42 @@ def __init__(self, name, action_spec, coro, loop=None):
         """ Cancel a particular goal's handler task. """
         task = self._cancel_cb(goal_handle)
         if task:
-            try:
```

Comment:
this behaviour was incorrect, suppressing an outer cancellation on the `cancel` coroutine.